csrc/polymorphic_value.h

```diff
@@ -163,14 +166,41 @@ inline Pointer operator+(int64_t offset, const Pointer& ptr) {
   return ptr + offset;
 }
 
+struct Opaque {
+  std::any value;
+
+  std::function<bool(const Opaque&, const Opaque&)> equals =
```

Comment:
Why is this variable necessary? Why can't `operator==` just compare addresses?

Reply:
Please see `ExprEvalTest.OpaqueEquality`